hw_14_sockets_client.py

```diff
@@ -0,0 +1,66 @@
+import datetime
+import socket
+from queue import LifoQueue
+from threading import Event, Thread
+from typing import Callable
+
+import keyboard
+import psutil
+
+
+class SensorMonitor:
+    """Performs threaded data acquisition from specified sensor."""
+
+    def __init__(self, source: Callable, interval: int):
+        self.source = source
+        self.interval = interval
+
+        self.data = LifoQueue()
+        self.keep_acq = Event()
+        self.keep_acq.set()
+
+    def start_collect(self):
+        Thread(target=self._continuous_acquisition, args=(self.source, self.interval)).start()
+
+    def _continuous_acquisition(self, source: Callable, interval: int):
+        while self.keep_acq.is_set():
+            self.data.put(source(interval))
+
+    def get_current_state(self) -> str:
+        return f'{datetime.datetime.now()}\tcpu_usage\t{self.data.get()}'
+
+    def cleanup(self):
```

Comment:
Could you demonstrate of working of `cleanup()` method?

Reply:
I'm not sure if I understood this part of task right.
The way I've done this, it clears monitor's object internal queue for data. It can be useful, in case queue length grows, which I mentioned in other conversation.

Reply:
Yes, it's right. You just haven't used it in your code.